crates/stages/src/stages/hashing_storage.rs

```diff
@@ -0,0 +1,491 @@
+
+                    let body = StoredBlockBody {
+                        start_tx_id: tx_id,
+                        tx_count: progress.body.len() as u64,
+                    };
+
+                    progress.body.iter().try_for_each(|transaction| {
+                        tx.put::<tables::TxHashNumber>(transaction.hash(), tx_id)?;
+                        tx.put::<tables::Transactions>(tx_id, transaction.clone())?;
+                        tx.put::<tables::TxTransitionIndex>(tx_id, transition_id)?;
+                        tx_id += 1;
+                        transition_id += 1;
+                        let (to, value) = match transaction.transaction {
+                            Transaction::Legacy(TxLegacy {
+                                to: TransactionKind::Call(to),
+                                value,
+                                ..
+                            }) => (to, value),
+                            _ => unreachable!(),
+                        };
+                        let new_entry =
+                            StorageEntry { key: keccak256("transfers"), value: U256::from(value) };
+                        self.insert_storage_entry(
+                            tx,
+                            (transition_id, to).into(),
+                            new_entry,
+                            progress.header.number == stage_progress,
+                        )
+                    })?;
+
+                    // Randomize rewards
+                    let has_reward: bool = rand::random();
+                    if has_reward {
+                        transition_id += 1;
+                        self.insert_storage_entry(
+                            tx,
+                            (transition_id, Address::random()).into(),
+                            StorageEntry {
+                                key: keccak256("mining"),
+                                value: U256::from(rand::random::<u32>()),
+                            },
+                            progress.header.number == stage_progress,
+                        )?;
+                    }
+
+                    tx.put::<tables::BlockTransitionIndex>(key.number(), transition_id)?;
+                    tx.put::<tables::BlockBodies>(key, body)
+                })?;
+            }
+
+            Ok(blocks)
+        }
+
+        fn validate_execution(
+            &self,
+            input: ExecInput,
+            output: Option<ExecOutput>,
+        ) -> Result<(), TestRunnerError> {
+            if let Some(output) = output {
+                let start_block = input.stage_progress.unwrap_or_default() + 1;
+                let end_block = output.stage_progress;
+                if start_block > end_block {
+                    return Ok(())
+                }
+            }
+            self.check_hashed_storage()
+        }
+
+        async fn after_execution(&self, _seed: Self::Seed) -> Result<(), TestRunnerError> {
```

Comment:
Is this used only for debugging or do we also want to add any asserts here?

Reply:
Yeah, there were debugs only prints, removed them.